databricks-sdk-java/src/main/java/com/databricks/sdk/core/oauth/DatabricksOAuthTokenSource.java

```diff
@@ -0,0 +1,200 @@
+
+import com.databricks.sdk.core.DatabricksException;
+import com.databricks.sdk.core.http.FormRequest;
+import com.databricks.sdk.core.http.HttpClient;
+import com.databricks.sdk.core.http.Response;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Strings;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Implementation of TokenSource that handles OAuth token exchange for Databricks authentication.
+ * This class manages the OAuth token exchange flow using ID tokens to obtain access tokens.
+ */
+public class DatabricksOAuthTokenSource implements TokenSource {
+  /** OAuth client ID used for token exchange */
+  private final String clientId;
+  /** Databricks account ID, used as audience if provided */
+  private final String accountId;
+  /** OpenID Connect endpoints configuration */
+  private final OpenIDConnectEndpoints endpoints;
+  /** Custom audience value for token exchange */
+  private final String audience;
+  /** Source of ID tokens used in token exchange */
+  private final IDTokenSource idTokenSource;
+  /** HTTP client for making token exchange requests */
+  private final HttpClient httpClient;
+
+  private DatabricksOAuthTokenSource(Builder builder) {
+    this.clientId = builder.clientId;
+    this.accountId = builder.accountId;
+    this.endpoints = builder.endpoints;
+    this.audience = builder.audience;
+    this.idTokenSource = builder.idTokenSource;
+    this.httpClient = builder.httpClient;
+  }
+
+  /**
+   * Builder class for constructing DatabricksOAuthTokenSource instances. Provides a fluent
+   * interface for setting required and optional parameters.
+   */
+  public static class Builder {
+    private final String clientId;
+    private final String host;
+    private final OpenIDConnectEndpoints endpoints;
+    private final IDTokenSource idTokenSource;
+    private final HttpClient httpClient;
+    private String accountId;
+    private String audience;
+
+    /**
+     * Validates that a value is non-empty and non-null for required fields.
+     *
+     * @param value The value to validate
+     * @param fieldName The name of the field being validated
+     * @throws IllegalArgumentException if validation fails
+     */
+    private static void validate(Object value, String fieldName) {
+      if (value == null) {
+        throw new IllegalArgumentException(fieldName + " must be non-null");
+      }
+      if (value instanceof String && ((String) value).isEmpty()) {
+        throw new IllegalArgumentException(fieldName + " must be non-empty");
+      }
+    }
+
+    /**
+     * Creates a new Builder with required parameters.
+     *
+     * @param clientId OAuth client ID
+     * @param host Databricks host URL
+     * @param endpoints OpenID Connect endpoints configuration
+     * @param idTokenSource Source of ID tokens
+     * @param httpClient HTTP client for making requests
```

Comment:
Let's document the error that this function can throw.
```suggestion
     * @param httpClient HTTP client for making requests
     * @throws IllegalArgumentException if any required parameter is null or empty
```